src/solvers/strings/array_pool.cpp

```diff
@@ -65,6 +65,10 @@ array_poolt::fresh_string(const typet &index_type, const typet &char_type)
   created_strings_value.insert(str);
   arrays_of_pointers.emplace(
     address_of_exprt(index_exprt(str, from_integer(0, index_type))), str);
+
+  // add length to length_of_array map
+  get_length(str);
```

Comment:
I wouldn't normally expect a `get_` function to have a side-effect like this, suggest renaming

Reply:
Would you be happy with `get_length_or_create`?

Reply:
Sounds good to me

Reply:
Renamed